projects/RabbitMQ.Client/util/DebugUtil.cs

```diff
@@ -99,55 +99,62 @@ public static void DumpKeyValue(string key, object value, TextWriter writer, int
         ///<summary>Dump properties of objects to the supplied writer.</summary>
         public static void DumpProperties(object value, TextWriter writer, int indent)
         {
-            if (value == null)
+            switch (value)
```

Comment:
Should we lock the csharp version to not accidentally use something that would not be compatible?